### PR TITLE
[MT-4872] Feature - New ImageNode model features

### PR DESCRIPTION
**Pull request by e1himself**
- Update `ImageNode` to support new `align` property
- Update `ImageNode` to support absolute sizing options in pixels
- No UI changes yet

### CODE REVIEW

**Review by aspirisen — APPROVED** [review]

File: packages/slate-editor/src/components/EditorBlock/Size.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 export type Size<U extends Unit = Unit> = { value: number; unit: U };
 
+export type SizeString = `${number}%` | `${number}px`;
```
Comment (aspirisen) [inline packages/slate-editor/src/components/EditorBlock/Size.ts:3]: 👍

File: packages/slate-types/src/nodes/ImageNode.ts [hunk 2]
```diff
@@ -11,16 +12,17 @@ export enum ImageLayout {
...
     file: UploadcareImageStoragePayload;
     /** empty string if no URL */
     href: string;
     layout: ImageLayout;
+    align: Alignment;
     new_tab: boolean;
-    /** matches this regexp: /^\d+(\.\d+)?%$/ */
-    width: string;
+    /** matches this regexp: /^\d+(\.\d+)?(%|px)$/ */
+    width: `${number}%` | `${number}px`;
```
Comment (OskarGasiorowski) [inline packages/slate-types/src/nodes/ImageNode.ts:25]: why not as `SizeString`?

**Review by OskarGasiorowski — APPROVED** [review]

File: packages/slate-types/src/nodes/ImageNode.ts (hunk 2 above)
Comment (e1himself) [reply packages/slate-types/src/nodes/ImageNode.ts]: You are right that this is the same type, but I didn't want to couple `Size` library module (`slate-editor` package) and this generic `ImageNode` interface (`slate-types` package).

I've deduplicated the type definition by exporting the `ImageWidth` type from the `@prezly/slate-types` package.

Comment (OskarGasiorowski) [reply packages/slate-types/src/nodes/ImageNode.ts]: oh, I haven't seen this is in two different packages. My bad :)

**Pull request merged**